src/gc-arena/src/types.rs

```diff
@@ -48,10 +61,41 @@ impl GcFlags {
         self.0.get() & 0x4 != 0x0
     }
 
+    /// This is `true` if we've traced a weak pointer during to this `GcBox`
+    /// during the most recent `Phase::Propagate`. This is reset back to
+    /// `false` during `Phase::Sweep`.
+    #[inline]
+    pub(crate) fn has_weak_ref(&self) -> bool {
```

Comment:
This name confused me for a while, it's not always set to true when a gc object *has* a weak ref, it's only used as a marker that's set during tracing. How about `trace_weak_ref` or something like that?

Reply:
I've renamed this to `traced_weak_ref`